### PR TITLE
feature:新增主机加锁与解锁插件

### PR DESCRIPTION
**Pull request by Tanghaohaohao**
- 新功能
    - 新增主机加锁与主机解锁两个插件

### CODE REVIEW

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py:1]: 新的插件的版本和模块不需要命名为 legacy，改为 v1.0

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class CmdbAddHostLockService(AddHostLockMixin, CCHostLockBaseService):
+    pass
+
+
+class CmdbAddHostLockComponent(Component):
+    name = _("主机加锁")
+    code = "cmdb_add_host_lock"
+    bound_service = CmdbAddHostLockService
+    form = "%scomponents/atoms/cc/cmdb_add_host_lock.js" % settings.STATIC_URL
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py:31]: version = "v1.0"

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+class CCHostLockBaseService(HostLockTypeService):
+    def inputs_format(self):
+        return [
+            self.InputItem(
+                name=_("主机 IP"),
+                key="cc_host_ip",
+                type="string",
+                schema=StringItemSchema(description=_("主机 IP，多个用英文逗号 `,` 分隔")),
+            ),
+            self.InputItem(
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py:59]: 不需要让用户输入云区域 ID

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        executor = parent_data.get_one_of_inputs("executor")
+        biz_cc_id = parent_data.get_one_of_inputs("biz_cc_id")
+
+        client = get_client_by_user(executor)
+        if parent_data.get_one_of_inputs("language"):
+            setattr(client, "language", parent_data.get_one_of_inputs("language"))
+            translation.activate(parent_data.get_one_of_inputs("language"))
+        cc_host_ip = data.get_one_of_inputs("cc_host_ip")
+        id_info = cc_get_ips_info_by_str(executor, biz_cc_id, cc_host_ip)
+        id_list = id_info["ip_result"] if id_info["result"] else []
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py:75]: 1. 这里应该获取 cc_get_ips_info_by_str 返回的 result 列表中字典的 HostID 字段，接口需要的是主机 ID
2. 需要判断是否有返回 invalid_ip，如果有，执行失败，并提示用户传入 ip 中有不在当前业务下的 ip

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+cc_handle_api_error = partial(handle_api_error, __group_name__)
+
+
+class HostLockTypeService(Service):
+    @abstractmethod
+    def host_lock_method(self):
+        raise NotImplementedError()
+
+
+class AddHostLockMixin(HostLockTypeService):
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py:40]: 关于子类特定行为的定义不应该放在 base 中

File: pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+
+## 标签
+`cc` `cmdb` `add_host_lock`
+
+## 参数说明
+
+* `biz_cc_id` 业务id
+
+* `cc_host_ip` 主机内网IP
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md:16]: 需要明确告知用户多个使用换行符分隔

File: pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# 主机加锁
+> 版本`legacy`
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md:2]: v1.0

File: pipeline_plugins/components/static/components/atoms/cc/cmdb_add_host_lock.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                    this._set_value($.context.getBkBizId())
+                }
+            }
+        },
+        {
+            tag_code: "cc_host_ip",
+            type: "textarea",
+            attrs: {
+                name: gettext("主机内网IP"),
+                placeholder: gettext("需要加锁的主机内网 IP，多个用英文逗号 `,` 分隔"),
```
Comment (homholueng) [inline pipeline_plugins/components/static/components/atoms/cc/cmdb_add_host_lock.js:46]: IP 使用换行符分隔

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py (hunk 1 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py:1]: done

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py (hunk 2 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/cc/host_lock/add_host_lock/legacy.py:31]: done

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py (hunk 3 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py:59]: done

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py (hunk 4 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py:75]: done

File: pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py (hunk 5 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/cc/host_lock/base.py:40]: done

File: pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md (hunk 6 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md:16]: done

File: pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md (hunk 7 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/collections/sites/open/docs/cc/add_host_lock/legacy.md:2]: done

File: pipeline_plugins/components/static/components/atoms/cc/cmdb_add_host_lock.js (hunk 8 above)
Comment (Tanghaohaohao) [reply pipeline_plugins/components/static/components/atoms/cc/cmdb_add_host_lock.js:46]: done

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-sops?pullRequest=2200) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-sops?pullRequest=2200) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-sops?pullRequest=2200) it.</sub>

**Pull request closed without merging**